dialogue-core/src/main/java/com/palantir/dialogue/core/PinUntilErrorChannel.java

```diff
@@ -104,9 +96,12 @@ static PinUntilErrorChannel from(
          * situation where there might be many nodes but all clients have decided to hammer one of them.
          */
         ImmutableList<LimitedChannel> initialShuffle = shuffleImmutableList(channels, random);
-        // We only rely on reference equality since we expect LimitedChannels to be reused across updates
-        int initialHost = initialShuffle.indexOf(initialChannel);
-        initialHost = initialHost == -1 ? 0 : initialHost;
+        int initialHost = 0;
+        if (initialChannel.isPresent()) {
+            // indexOf relies on reference equality since we expect LimitedChannels to be reused across updates
+            initialHost = initialShuffle.indexOf(initialChannel.get());
```

Comment:
nit, might be easier to read as one line using max instead of two asignments
```suggestion
            initialHost = Math.max(0, initialShuffle.indexOf(initialChannel.get()));
```